edweiss-firebase/functions/src/utils/custom-sanitizer/quiz.ts

```diff
@@ -65,4 +66,68 @@ export namespace CustomPredicateQuiz {
 			)]
 		})
 
-}
\ No newline at end of file
+	export const isValidLectureQuiz: Predicate<LectureQuizzes.LectureQuiz> =
+
+		Predicate.fields<LectureQuizzes.LectureQuiz>({
+			answer: Predicate.dispatch("type", {
+				MCQAnswersIndices: Predicate.fields({
+					type: Predicate.is("MCQAnswersIndices"),
+					value: Predicate.forEach(Predicate.isNumber)
+				}),
+				TFAnswer: Predicate.fields({
+					type: Predicate.is("TFAnswer"),
+					value: Predicate.isOptionalBoolean
+				}),
+
+
+			}),
+			ended: Predicate.isBoolean,
+			exercise: Predicate.dispatch("type", {
+				MCQ: Predicate.fields({
+					answersIndices: Predicate.forEach(Predicate.isNumber),
+					imageURL: Predicate.isOptionalString,
+					numberOfAnswers: Predicate.isStrictlyPositive,
+					propositions: Predicate.forEach(
+						Predicate.fields<Quizzes.MCQProposition>({
+							description: Predicate.isString,
+							id: Predicate.isPositive,
+							type: Predicate.is("MCQProposition"),
+							imageUrl: Predicate.isOptionalString
+						})
+					),
+					question: Predicate.isString
+				}),
+				TF: Predicate.fields({
+					answer: Predicate.isBoolean,
+					question: Predicate.isString,
+					imageURL: Predicate.isOptionalString
+				})
+			}),
+			showResultToStudents: Predicate.isBoolean
+		});
+
+	export const isValidLectureQuizAttempt: Predicate<LectureQuizzesAttempts.LectureQuizAttempt> =
+
+		Predicate.dispatch("type", {
+			MCQAnswersIndices: Predicate.fields({
+				type: Predicate.is("MCQAnswersIndices"),
+				value: Predicate.forEach(Predicate.isNumber)
+			}),
+			TFAnswer: Predicate.fields({
+				type: Predicate.is("TFAnswer"),
+				value: Predicate.isOptionalBoolean
+			})
+		})
+
+	export const isValidLectureQuizEvent: Predicate<LectureDisplay.QuizLectureEvent> =
+		Predicate.fields<LectureDisplay.QuizLectureEvent>({
+			done: Predicate.isBoolean,
+			pageNumber: Predicate.isPositive,
```

Comment:
The maximum number of pages is exterior to the lecture quiz: for now create a Lecture event with hard coded values (page 0 etc.). I do not know the exact implementation of slide creation for now (and it might even stay hard coded for the final demo). This will change once I see the context of slide creation in detail with @Gutogmaia and @adamm-alaoui-pro .